WalletWasabi.Fluent/ViewModels/Wallets/Send/TransactionPreviewViewModel.cs

```diff
@@ -22,75 +27,255 @@ public partial class TransactionPreviewViewModel : RoutableViewModel
+
 				NextCommand = ReactiveCommand.CreateFromTask(async () =>
 				{
-					var saved = await TransactionHelpers.ExportTransactionToBinaryAsync(transaction);
+					var saved = await TransactionHelpers.ExportTransactionToBinaryAsync(_transaction);
 
 					if (saved)
 					{
 						Navigate().To(new SuccessViewModel("The PSBT has been successfully created."));
 					}
 				});
+
 				_nextButtonText = "Save PSBT file";
 			}
 			else
 			{
-				NextCommand = ReactiveCommand.CreateFromTask(async () => await OnConfirmAsync(transaction));
+				NextCommand = ReactiveCommand.CreateFromTask(async () => await OnConfirmAsync(_transaction));
+
 				_nextButtonText = "Confirm";
 			}
 
+			AdjustFeeCommand = ReactiveCommand.CreateFromTask(async () =>
+			{
+				var feeRateDialogResult = await NavigateDialogAsync(new SendFeeViewModel(wallet, info, false));
+
+				if (feeRateDialogResult.Kind == DialogResultKind.Normal)
+				{
+					_info.FeeRate = feeRateDialogResult.Result;
+				}
+
+				var newTransaction = await BuildTransactionAsync();
+
+				if (newTransaction is { })
+				{
+					UpdateTransaction(newTransaction);
+				}
+			});
+
+			AvoidChangeCommand = ReactiveCommand.CreateFromTask(async () =>
+			{
+				var optimisePrivacyDialog =
+					await NavigateDialogAsync(new OptimisePrivacyViewModel(wallet, info, _transaction!));
+
+				if (optimisePrivacyDialog.Kind == DialogResultKind.Normal && optimisePrivacyDialog.Result is { })
+				{
+					UpdateTransaction(optimisePrivacyDialog.Result);
+				}
+			});
 		}
 
 		public bool PreferPsbtWorkflow => _wallet.KeyManager.PreferPsbtWorkflow;
 
-		public string AmountText { get; }
-
 		public string AddressText { get; }
 
-		public string FeeText { get; }
-
 		public string? PayJoinUrl { get; }
 
 		public bool IsPayJoin { get; }
 
+		public ICommand AdjustFeeCommand { get; }
+
+		public ICommand AvoidChangeCommand { get; }
+
+		private async Task InitialiseTransactionAsync()
+		{
+			var transactionInfo = _info;
+			var targetAnonymitySet = _wallet.ServiceConfiguration.GetMixUntilAnonymitySetValue();
+			var mixedCoins = _wallet.Coins.Where(x => x.HdPubKey.AnonymitySet >= targetAnonymitySet).ToList();
+			var totalMixedCoinsAmount = Money.FromUnit(mixedCoins.Sum(coin => coin.Amount), MoneyUnit.Satoshi);
+
+			transactionInfo.Coins = mixedCoins;
+
+			var feeDialogResult = await NavigateDialogAsync(new SendFeeViewModel(_wallet, transactionInfo, true));
+
+			if (feeDialogResult.Kind == DialogResultKind.Normal)
+			{
+				transactionInfo.FeeRate = feeDialogResult.Result;
+			}
+
+			if (transactionInfo.Amount > totalMixedCoinsAmount)
+			{
+				var privacyControlDialogResult =
+					await NavigateDialogAsync(new PrivacyControlViewModel(_wallet, transactionInfo));
+
+				if (privacyControlDialogResult.Kind == DialogResultKind.Normal &&
+				    privacyControlDialogResult.Result is { })
+				{
+					transactionInfo.Coins = privacyControlDialogResult.Result;
```

Comment:
// TODO in this case or with hw wallet??? we must show them the change optimisation screen.